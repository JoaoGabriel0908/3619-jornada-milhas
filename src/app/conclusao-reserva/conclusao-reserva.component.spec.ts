import { ComponentFixture, TestBed } from '@angular/core/testing';
import { ConclusaoReservaComponent } from './conclusao-reserva.component';
import { HttpClientTestingModule } from '@angular/common/http/testing';

describe('ConclusaoReservaComponent', () => {
  let component: ConclusaoReservaComponent;
  let fixture: ComponentFixture<ConclusaoReservaComponent>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [
        HttpClientTestingModule, // Ensure HttpClientTestingModule is imported if needed
      ],
      declarations: [ConclusaoReservaComponent]
    });
    fixture = TestBed.createComponent(ConclusaoReservaComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('deve exibir o titulo "Reserva concluída com sucesso."', () => {
    const compiled = fixture.nativeElement;
    expect(compiled.querySelector('h1.confirmation-title').textContent).toContain('Reserva concluída com sucesso.')
  })

  it('deveria renderizar o conteúdo baseado nas propriedades alt e src', () => {
    const compiled = fixture.nativeElement;
    const imgElement = compiled.querySelector('.confirmation-image');
    imgElement.src = '../../assets/imagens/conclusao-reserva.png';
    imgElement.alt = 'Reservation confirmed illustration';

    expect(fixture.nativeElement).toMatchSnapshot();
  })
});
