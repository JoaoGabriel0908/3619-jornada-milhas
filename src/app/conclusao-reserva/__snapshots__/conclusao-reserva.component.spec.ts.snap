// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConclusaoReservaComponent deveria renderizar o conteúdo baseado nas propriedades alt e src 1`] = `
<div
  id="root2"
>
  <app-banner
    alt="Banner conclusão"
    src="assets/imagens/banner-conclusão.png"
  />
  <app-container>
    <div
      class="reservation-confirmation"
      style="height: 100%;"
    >
      <header
        class="header"
      >
        <nav
          class="back-nav"
        >
          <img
            alt="Back arrow"
            class="back-icon"
            src="../../assets/icones/back-icon.svg"
          />
          <a
            appbackbutton=""
            class="back-link"
          >
            Voltar
          </a>
        </nav>
      </header>
      <section
        class="confirmation-content"
      >
        <img
          alt="Reservation confirmed illustration"
          class="confirmation-image"
          src="../../assets/imagens/conclusao-reserva.png"
        />
        <h1
          class="confirmation-title"
        >
          Reserva concluída com sucesso.
        </h1>
        <p
          class="confirmation-message"
        >
           Uhuuuul, prepare suas malas pois a compra foi confirmada e o seu próximo destino já está programado! 
        </p>
      </section>
    </div>
  </app-container>
</div>
`;
